libs/prelude/Prelude/Types.idr

```diff
@@ -875,6 +882,67 @@ fastUnpack : String -> List Char
 -- always use 'fastPack' at run time
 %transform "fastUnpack" unpack = fastUnpack
 
+namespace String
+  public export
+  (++) : (x : String) -> (y : String) -> String
+  x ++ y = strInduction y (\c, s, res => strCons c res) x
+
+  %transform "fastAppend" String.(++) = prim__strAppend
```

Comment:
Are we sure this rule will always fire? For all we know the compiler may decide to inline `++`
(the definition is not recursive & not pattern-matching) and then the transform rule will not
be deployed on the `strInduction`-headed expression.

Same comment for `reverse`, `substr`, etc.